netbox/dcim/models/device_components.py

```diff
@@ -904,6 +904,12 @@ def clean(self):
                 ).format(untagged_vlan=self.untagged_vlan)
             })
 
+        # Validate that tagged-all payload does not include tagged_vlans
+        if self.mode != InterfaceModeChoices.MODE_TAGGED and self.tagged_vlans:
+            raise ValidationError({
+                'tagged_vlans': "Interface mode does not support including tagged vlans"
+            })
+
```

Comment:
We should move this to `clean()` on BaseInterface to ensure it applies to VM interfaces as well.